torch/distributed/_tensor/_xla.py

```diff
@@ -0,0 +1,201 @@
+
+      # `placements = [Replicate(), Shard(0)]` describes sharding per mesh dim,
+      # and this is equivalent to `partition_spec = (1, None, None)` which is
+      # sharding per input tensor dimension.
+      partition_spec = convert_to_xla_partition_spec(t, placements)
+      >> (1, None, None)
+      ```
+    """
+    # per tensor dimension sharding
+    sharding_spec = [None] * len(tensor.shape)
+    for mesh_idx, spec in enumerate(placements):
+        if spec.is_shard():  # type:ignore[truthy-function]
+            # mesh_idx to tensor_idx (spec.dim)
+            tensor_idx = spec.dim  # type:ignore[attr-defined]
+            sharding_spec[tensor_idx] = mesh_idx  # type:ignore[call-overload]
+        elif spec.is_replicate():
+            # spec.dim is already set to None by default
+            continue
+        else:
+            raise ValueError(f"Unsupported placement type: {type(spec).__name__}")
+    return tuple(sharding_spec)  # type:ignore[return-value]
+
+
+@with_xla
+def xla_distribute_tensor(
+    tensor: torch.Tensor,
+    device_mesh: Optional[DeviceMesh] = None,
+    placements: Optional[Sequence[Placement]] = None,
+) -> "XLAShardedTensor":
+    """
+    Distribute a torch.Tensor to the `device_mesh` according to the `placements`
+    specified. The rank of `device_mesh` and `placements` must be the same.
+
+    Args:
+        tensor (torch.Tensor): torch.Tensor to be distributed. Note that if you
+            want to shard a tensor on a dimension that is not evenly divisible by
+            the number of devices in that mesh dimension, we use `torch.chunk`
+            semantic to shard the tensor and scatter the shards.
+        device_mesh (:class:`DeviceMesh`, optional): DeviceMesh to distribute the
+            tensor, if not specified, must be called under a DeviceMesh context
+            manager, default: None
+        placements (List[:class:`Placement`], optional): the placements that
+            describes how to place the tensor on DeviceMesh, must have the same
+            number of elements as `device_mesh.ndim`. If not specified, we will
+            by default replicate the tensor across the `device_mesh` from the
+            first rank of each dimension of the `device_mesh`.
+
+    Returns:
+        A :class:`XLAShardedTensor` object
+
+    .. note:: We return a XLAShardedTensor with a global view and access to local shards.
+    The successive ops would be programmed as if on a single-device and without calling
+    any explicit collective ops. This is different from returning a DTensor with
+    local shard. The actual sharded computation on the sharding annotated tensor
+    happens lazily, is transparent to the user. In the future, we will introduce
+    a new DTensor type for this kind of programming-mode (single-controller) and return.
+    """
+    # get default device mesh if there's nothing specified
+    dt_mesh = device_mesh or mesh_resources.get_current_mesh()
```

Comment:
nit: this is already been called in the `distribute_tensor` API and `xla_distribute_tensor` calls after that, so we can safely remove this line and assume `device_mesh` is not optional

Reply:
Done